src/apps/compound/positions.ts

```diff
@@ -0,0 +1,173 @@
+} from './abis/compound-multicall'
+
+// Data from https://docs.compound.finance/
+const MARKETS: { networkId: NetworkId; address: Address }[] = [
+  {
+    // USDC
+    networkId: NetworkId['op-mainnet'],
+    address: '0x2e44e174f7D53F0212823acC11C01A11d58c5bCB',
+  },
+  {
+    // USDT
+    networkId: NetworkId['op-mainnet'],
+    address: '0x995E394b8B2437aC8Ce61Ee0bC610D617962B214',
+  },
+  {
+    // USDC
+    networkId: NetworkId['ethereum-mainnet'],
+    address: '0xc3d688B66703497DAA19211EEdff47f25384cdc3',
+  },
+  {
+    // WETH
+    networkId: NetworkId['ethereum-mainnet'],
+    address: '0xA17581A9E3356d9A858b789D68B4d866e593aE94',
+  },
+  {
+    // USDC
+    networkId: NetworkId['arbitrum-one'],
+    address: '0x9c4ec768c28520B50860ea7a15bd7213a9fF58bf',
+  },
+  {
+    // WETH
+    networkId: NetworkId['arbitrum-one'],
+    address: '0x6f7D514bbD4aFf3BcD1140B7344b32f063dEe486',
+  },
+  {
+    // USDT
+    networkId: NetworkId['arbitrum-one'],
+    address: '0xd98Be00b5D27fc98112BdE293e487f8D4cA57d07',
+  },
+]
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'compound',
+      name: 'Compound',
+      description: 'Compound markets',
+    }
+  },
+  async getPositionDefinitions(
+    networkId: NetworkId,
+    address: Address | undefined,
+  ) {
+    const markets = MARKETS.filter((market) => market.networkId === networkId)
+    if (!markets.length || !address) {
+      return []
+    }
+
+    const client = getClient(networkId)
+
+    const results = await client.readContract({
+      code: compoundMulticallBytecode,
+      abi: compoundMulticallAbi,
+      functionName: 'getUserPositions',
+      args: [address, markets.map((m) => m.address)],
+    })
+
+    return results.flatMap(
+      ({ baseToken, baseTokenBalance, borrowBalance, collaterals }, index) => {
+        const balances: {
+          token: Address
+          market: Address
+          balance: bigint
+          type: string
+        }[] = [
+          {
+            token: baseToken,
+            market: markets[index].address,
+            balance: baseTokenBalance,
+            type: 'Supply',
+          },
+          {
+            token: baseToken,
+            market: markets[index].address,
+            balance: -borrowBalance,
+            type: 'Debt',
+          },
+        ]
+
+        for (const collateral of collaterals) {
+          balances.push({
+            token: collateral.asset,
+            market: markets[index].address,
+            balance: collateral.balance,
+            type: 'Collateral',
+          })
+        }
+
+        const positions: ContractPositionDefinition[] = []
+
+        for (const { token, balance, market, type } of balances) {
+          if (balance === 0n) {
+            continue
+          }
+          positions.push({
+            type: 'contract-position-definition',
+            networkId,
+            address: `${market.toLowerCase()}-${token.toLowerCase()}-${type}`,
```

Comment:
Ah yes I really need to introduce a proper position identifier so we don't need this workaround.
This has been a problem for uniswap v3 and other apps too.
Related to shortcuts and the new earn feature in Valora, this is also on my TODO list for this week.

I thought Valora would still display individual positions, even with the same address, but maybe we broke that recently.